crates/oracle/src/oracle.rs

```diff
@@ -304,13 +239,70 @@ fn registered_networks(
     }
 }
 
-fn latest_blocks_to_message(latest_blocks: BTreeMap<Caip2ChainId, BlockPtr>) -> ee::Message {
-    Message::SetBlockNumbersForNextEpoch(
+fn set_block_numbers_for_next_epoch(
+    subgraph_state: &SubgraphStateTracker<SubgraphQuery>,
+    latest_blocks: BTreeMap<Caip2ChainId, BlockPtr>,
+) -> Vec<u8> {
+    let registered_networks = registered_networks(subgraph_state);
+
+    let mut block_nums_for_next_epoch = BTreeMap::new();
+    let mut ignored_networks = BTreeSet::new();
+    for (chain_id, block_num) in &latest_blocks {
+        if registered_networks
+            .iter()
+            .any(|network| &network.id == chain_id)
+        {
+            block_nums_for_next_epoch.insert(chain_id, block_num);
+        } else {
+            ignored_networks.insert(chain_id);
```

Comment:
This `BTreeSet` is not being used after this insertion.

I think we should log its contents as a warning _(if any)_, as it would indicate that the Oracle is unnecessarily pulling blocks from unregistered chains.

Depending on our interpretation, this could also be seen as an error, which was previously represented by the `Error::MalconfiguredIndexedChains` variant.

Reply:
I think a warning is enough. With an error, we wouldn't be able to ever post block updates if the configuration contains any unregistered networks. I think that's excessively restrictive.

Fixed in https://github.com/graphprotocol/block-oracle/pull/194/commits/bf80c4e9a3060643788356f7afb10ad470fc3505.